### PR TITLE
refactor(experimental): use new RPC libraries in @solana/rpc-graphql

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR rewires the RPC dependencies of `@solana/rpc-graphql` to use the new RPC packages.

RPC packages reshuffle@2x (13)

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2156?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#2158** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2158?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2157** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2157?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2156** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2156?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2155** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2155?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2154** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2154?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2153** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2153?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2152** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2152?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2151** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2151?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2150** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2150?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2149** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2149?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2148** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2148?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2147** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2147?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2146** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2145** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2144** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2156?utm_source=stack-comment).

File: packages/rpc-graphql/src/__tests__/account-test.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
     GetAccountInfoApi,
     GetBlockApi,
     GetProgramAccountsApi,
     GetTransactionApi,
-} from '@solana/rpc-core';
-import { createHttpTransport, createJsonRpc } from '@solana/rpc-transport';
-import type { Rpc } from '@solana/rpc-types';
+    Rpc,
+} from '@solana/rpc';
+import { createHttpTransport } from '@solana/rpc-transport-http';
```
Comment (buffalojoec) [inline packages/rpc-graphql/src/__tests__/account-test.ts:10]: Up to you, but I think `createDefaultHttpTransport` from `@solana/rpc` would work here, and then the transport lib isn't needed anywhere. The rest of the library should only need `@solana/rpc`.

File: packages/rpc-graphql/package.json [hunk 2]
```diff
@@ -73,8 +73,9 @@
         "@solana/build-scripts": "workspace:*",
         "@solana/eslint-config-solana": "^1.0.2",
         "@solana/keys": "workspace:*",
-        "@solana/rpc-core": "workspace:*",
-        "@solana/rpc-transport": "workspace:*",
+        "@solana/rpc": "workspace:*",
+        "@solana/rpc-subscriptions": "workspace:*",
```
Comment (buffalojoec) [inline packages/rpc-graphql/package.json:77]: The subscriptions package shouldn't be needed either.

**Review by buffalojoec — APPROVED** [review]
I love how your naming conventions allowed this to remain Solana-specific (a.k.a. I don't have to do anything).

File: packages/rpc-graphql/src/__tests__/account-test.ts (hunk 1 above)
Comment (lorisleiva) [reply packages/rpc-graphql/src/__tests__/account-test.ts:10]: That's a good shout, I'll update accordingly.

File: packages/rpc-graphql/package.json (hunk 2 above)
Comment (lorisleiva) [reply packages/rpc-graphql/package.json:77]: Hmm not sure why but I thought it was imported somewhere in this package haha. Thanks for raising this, I'll remove the dependency.

**Comment by lorisleiva** [comment]
✅ This PR was merged as part of https://github.com/solana-labs/solana-web3.js/pull/2168.

**Pull request closed without merging**